homeassistant/components/gpstracker/config_flow.py

```diff
@@ -0,0 +1,135 @@
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+from .helpers import get_gpstracker_client
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    cfg = gps_tracker.Config(  # type: ignore[call-arg]
+        password=data[CONF_PASSWORD],
+        username=data[CONF_USERNAME],
+    )
+
+    async with get_gpstracker_client(hass, cfg) as client:
+        try:
+            await client.get_devices()
+        except gps_tracker.client.exceptions.UnauthorizedQuery as err:
+            raise InvalidAuth from err
+        except (
+            gps_tracker.client.exceptions.HttpException,
+            aiohttp.ClientResponseError,
+        ) as err:
+            raise CannotConnect from err
+
+
+class GpsTrackerFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for GPS Tracker."""
+
+    VERSION = 1
+
+    entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+            await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
+            self._abort_if_unique_id_configured()
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(
+                title=user_input[CONF_USERNAME], data=user_input
+            )
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication of GPS Tracker."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication."""
+        errors = {}
+
+        if user_input is not None and self.entry:
```

Comment:
I think it would be clearer to use the same pattern as  async_step_user.
```suggestion
        if user_input is None:
```
Return early and unindent below

Reply:
Fixed in 24518a7f076a94df654839d18d75a5cf1e65b362